File: examples/hello.kt
// Kotlin Example - Modern JVM language

fun main() {
    println("🎯 Hello from Kotlin!")
    
    // Variables
    val message = "Kotlin is concise and safe!"
    val number = 42
    
    println("Message: $message")
    println("Number: $number")
    
    // List
    val numbers = listOf(1, 2, 3, 4, 5)
    println("Numbers: ${numbers.joinToString(", ")}")
    
    // Function call
    val result = add(10, 20)
    println("10 + 20 = $result")
    
    // Data class
    val person = Person("John", 30)
    println("\nPerson: $person")
    
    // Lambda
    val doubled = numbers.map { it * 2 }
    println("Doubled: ${doubled.joinToString(", ")}")
}

fun add(a: Int, b: Int): Int {
    return a + b
}

data class Person(val name: String, val age: Int)

File: examples/hello.scala
// Scala Example - Functional + Object-oriented

object Hello {
  def main(args: Array[String]): Unit = {
    println("⚡ Hello from Scala!")
    
    // Variables
    val message = "Scala is powerful and elegant!"
    val number = 42
    
    println(s"Message: $message")
    println(s"Number: $number")
    
    // List
    val numbers = List(1, 2, 3, 4, 5)
    println(s"Numbers: ${numbers.mkString(", ")}")
    
    // Function call
    val result = add(10, 20)
    println(s"10 + 20 = $result")
    
    // Map
    val doubled = numbers.map(_ * 2)
    println(s"Doubled: ${doubled.mkString(", ")}")
    
    // Case class
    case class Person(name: String, age: Int)
    val person = Person("John", 30)
    println(s"\nPerson: $person")
  }
  
  def add(a: Int, b: Int): Int = a + b
}
